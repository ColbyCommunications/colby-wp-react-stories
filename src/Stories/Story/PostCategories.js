import React from 'react';
import PropTypes from 'prop-types';
import styled from 'styled-components';

import { withStateContext, withActionContext } from '../Context';

const StyledContainer = styled.div`
  display: flex;
  flex-wrap: wrap;
  padding: 1rem;
`;

const StyledButton = styled.button`
  flex: 0 0 46%;
  padding: 0.375rem;
  margin-right: 0.25rem;
  margin-bottom: 0.5rem;
  margin-left: 0.25rem;
  font-size: 66.6%;
  line-height: 1.2;
  color: black;
  color: var(--color, black);
  text-transform: uppercase;
  cursor: pointer;
  background: rgba(0, 0, 0, 0.075);
  background: var(--light-gray, rgba(0, 0, 0, 0.075));
  border: 1px solid rgba(0, 0, 0, 0.15);
  border-radius: 2px;
  transition: 0.2s background-color;

  &:hover {
    background: rgba(0, 0, 0, 0.1);
    background: var(--background-hover-color, rgba(0, 0, 0, 0.1));
  }
`;

export const PostCategories = ({
  post, setActiveCategory, categories,
}) => (
  <StyledContainer>
    {categories
      .filter(category => post.categories.indexOf(category.id) !== -1)
      .map(({ id, name, slug }) => (
        <StyledButton
          className={`Stories__category--${slug}`}
          key={id}
          onClick={() => {
            setActiveCategory(String(id));
          }}
          dangerouslySetInnerHTML={{ __html: name }}
        />
        ))}
  </StyledContainer>
);

PostCategories.propTypes = {
  post: PropTypes.objectOf(PropTypes.any).isRequired,
  setActiveCategory: PropTypes.func.isRequired,
  categories: PropTypes.arrayOf(PropTypes.object).isRequired,
};

export default withStateContext(withActionContext(PostCategories));

