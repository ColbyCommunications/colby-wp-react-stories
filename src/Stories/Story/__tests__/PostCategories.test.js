import React from 'react';
import renderer from 'react-test-renderer';
import Enzyme, { mount } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import sinon from 'sinon';

import { PostCategories } from '../PostCategories';
import { categories, posts } from '../../__tests__/testData';

Enzyme.configure({ adapter: new Adapter() });

describe('PostCategories', () => {
  it('renders', () => {
    const tree = renderer.create(<PostCategories
      post={posts[0]}
      categories={categories}
      setActiveCategory={() => null}
    />).toJSON();

    expect(tree).toMatchSnapshot();
  });

  it('handles a click', () => {
    const setActiveCategory = sinon.spy();
    const wrapper = mount(<PostCategories
      post={posts[0]}
      categories={categories}
      setActiveCategory={setActiveCategory}
    />);

    const button = wrapper.find('button');
    button.simulate('click');

    expect(setActiveCategory.calledOnce).toEqual(true);
    expect(setActiveCategory.calledWith(String(posts[0].categories[0]))).toEqual(true);
  });
});
