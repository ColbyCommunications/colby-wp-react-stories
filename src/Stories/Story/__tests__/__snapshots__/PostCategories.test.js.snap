// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PostCategories renders 1`] = `
<div
  className="sc-bdVaJa fEpVJS"
>
  <button
    className="Stories__category--undefined sc-bwzfXH eUaYFq"
    dangerouslySetInnerHTML={
      Object {
        "__html": "My Category",
      }
    }
    onClick={[Function]}
  />
</div>
`;
